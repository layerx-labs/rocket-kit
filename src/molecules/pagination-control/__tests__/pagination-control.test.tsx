import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { PaginationControl } from '../../..';

describe('PaginationControl', () => {
  it('renders', () => {
    const { asFragment } = render(
      <PaginationControl page={0} pageCount={10} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
