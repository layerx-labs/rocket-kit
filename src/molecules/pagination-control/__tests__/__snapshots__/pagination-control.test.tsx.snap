// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationControl renders 1`] = `
<DocumentFragment>
  .c0 {
  --size: 1.875rem;
}

.c0 ul.pagination {
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
  list-style: none;
}

.c0 ul.pagination li {
  min-width: var(--size);
  height: var(--size);
  cursor: pointer;
}

.c0 ul.pagination li.previous,
.c0 ul.pagination li.next {
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c0 ul.pagination li.previous a svg,
.c0 ul.pagination li.next a svg {
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c0 ul.pagination li a,
.c0 ul.pagination li .break {
  min-width: var(--size);
  height: var(--size);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c0 ul.pagination li a {
  border-radius: 0.5rem;
  min-width: var(--size);
  height: var(--size);
  padding: 0 0.3125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #939393;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 ul.pagination li a:hover {
  color: #000000;
}

.c0 ul.pagination li a svg {
  width: 1.125rem;
  height: auto;
  fill: #4250e4;
}

.c0 ul.pagination li.active {
  pointer-events: none;
}

.c0 ul.pagination li.active a {
  background-color: #4250e4;
  color: #ffffff;
}

.c0 ul.pagination.dark a {
  color: #c4c7d3;
}

.c0 ul.pagination.dark a:hover {
  color: #ffffff;
}

<div
    class="c0"
  >
    <ul
      class="pagination"
    >
      <li
        class="previous disabled"
      >
        <a
          aria-disabled="true"
          aria-label="Previous page"
          rel="prev"
          role="button"
          tabindex="0"
        >
          <svg
            viewBox="0 0 32 32"
          >
            <path
              d="M22.6986 26.027C22.0701 26.7812 20.9491 26.8832 20.1948 26.2546L9.52814 17.3657C9.12282 17.0279 8.88847 16.5276 8.88847 16C8.88847 15.4724 9.12282 14.972 9.52814 14.6342L20.1948 5.74536C20.9491 5.1168 22.0701 5.21871 22.6986 5.97298C23.3272 6.72725 23.2253 7.84826 22.471 8.47682L13.4432 16L22.471 23.5231C23.2253 24.1517 23.3272 25.2727 22.6986 26.027Z"
            />
          </svg>
        </a>
      </li>
      <li
        class="page active"
      >
        <a
          aria-current="page"
          aria-label="Page 1 is your current page"
          role="button"
          tabindex="0"
        >
          1
        </a>
      </li>
      <li
        class="page"
      >
        <a
          aria-label="Page 2"
          role="button"
          tabindex="0"
        >
          2
        </a>
      </li>
      <li
        class="page"
      >
        <a
          aria-label="Page 3"
          role="button"
          tabindex="0"
        >
          3
        </a>
      </li>
      <li
        class="page"
      >
        <a
          aria-label="Page 4"
          role="button"
          tabindex="0"
        >
          4
        </a>
      </li>
      <li
        class="break"
      >
        <a
          class="break"
          role="button"
          tabindex="0"
        >
          ...
        </a>
      </li>
      <li
        class="page"
      >
        <a
          aria-label="Page 10"
          role="button"
          tabindex="0"
        >
          10
        </a>
      </li>
      <li
        class="next"
      >
        <a
          aria-disabled="false"
          aria-label="Next page"
          rel="next"
          role="button"
          tabindex="0"
        >
          <svg
            viewBox="0 0 32 32"
          >
            <path
              d="M9.30048 5.97303C9.92904 5.21876 11.05 5.11685 11.8043 5.74541L22.471 14.6343C22.8763 14.9721 23.1106 15.4724 23.1106 16C23.1106 16.5276 22.8763 17.028 22.471 17.3658L11.8043 26.2546C11.05 26.8832 9.92904 26.7813 9.30048 26.027C8.67192 25.2727 8.77383 24.1517 9.5281 23.5232L18.5559 16L9.5281 8.47686C8.77383 7.8483 8.67192 6.7273 9.30048 5.97303Z"
            />
          </svg>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
